stackslib/src/net/relay.rs

```diff
@@ -2049,9 +2049,11 @@ impl Relayer {
             }
         }
         if !http_uploaded_blocks.is_empty() {
-            coord_comms.inspect(|comm| {
-                comm.announce_new_stacks_block();
-            });
+            if let Some(comm) = coord_comms {
```

Comment:
Please revert this.  `.inspect()` is meant to reduce the indentation level required to do something on the `T` in `Option<T>`.

Reply:
Hm, isn't this the same indentation level (barring the new `if !comm.announce...`)? Also, `inspect` isn't able to propagate the error up. I could use `and_then`, but that requires a result of `Option<T>`, so I'd have to chain that with some other code to bubble up the error.

Reply:
This function doesn't return an error; it instead only returns whether or not the `stop` flag has been raised to the chains coordinator (but note that the chains coordinator thread may still be running).  `announce_new_stacks_block()` is a best-effort function -- there's really no way for the caller to meaningfully act on its return value; the return value is there as a hint to the caller about the (impending) shutdown, but its return value doesn't say anything about whether or not the coordinator thread will process the newly-stored Stacks block prior to exiting.

Reply:
Sorry for the miscommunication - yeah I know `announce_new_stacks_block` only returns `bool`, I meant the "wrapper" around `Option<CoordinatorChannels>`. Here's the same exact code used elsewhere in the file which uses the `if let Some(...` pattern:

https://github.com/stacks-network/stacks-core/blob/8462f901d4cbd258e99b5176e8f23d3e320077f7/stackslib/src/net/relay.rs#L985-L989

Using `inspect` doesn't work here because `inspect` doesn't return any type from the lambda, and things like `and_then` need to return `Option<T>`, so you can't do an early return like you can with `if let Some`.

Also, just to be clear - the point of this change is to return an `Err` if `announce_new_stacks_block` returns false, instead of just ignoring the result. If _that_ functional change isn't desired, this whole diff can indeed be reverted!

Reply:
> Also, just to be clear - the point of this change is to return an Err if announce_new_stacks_block returns false, instead of just ignoring the result. If that functional change isn't desired, this whole diff can indeed be reverted!

This is what I'm saying.  The caller has no realistic way to handle the situation where `announce_new_stacks_block()` returns `false`, so it's fine to ignore its return value.

Reply:
Ok, reverted!